crates/goose/src/hints/load_hints.rs

```diff
@@ -7,10 +7,31 @@ use std::{
 use crate::config::paths::Paths;
 use crate::hints::import_files::read_referenced_files;
 
+/// Build a gitignore matcher for the given directory
+pub fn build_gitignore(working_dir: &Path) -> Gitignore {
```

Comment:
I don't think we want to take gitignore into account, there was a recent change to address that as it was incorrect